### DISCUSSION TITLE
gdb - running tests - need help with debugging

### DISCUSSION THREAD

**Discussion by palashahuja** [libuv/libuv#4095] [General]
I tried debugging the following libuv program run (running from the root folder)
```
gdb --args build/uv_run_tests_a strscpy
```
And then if I set the breakpoint at `uv__strscpy ` and run the program with the following commands
```
set follow-fork-mode child
break uv__strscpy
run
lay next
```
After this if I do `next` it fails with the following error
```
Thread 2.1 "uv_run_tests_a" received signal SIGTERM, Terminated.
uv__strscpy (d=0xbefff1e4 "x\230\016", s=0x103264 "", n=0)
    at /home/ahujapalash/Documents/repos/libuv/libuv-1.45.0/test/../src/strscpy.c:32
```

**Comment by vtjnash** [comment]
Write the test name twice in the call to avoid the SIGTERM from the test harness

**Comment by bnoordhuis** [comment]
Note that some tests need helper processes:
```
$ grep TEST_HELPER test/test-list.h
  TEST_HELPER (tcp_ping_pong, tcp4_echo_server)
  TEST_HELPER (tcp_ping_pong_vec, tcp4_echo_server)
  TEST_HELPER (tcp6_ping_pong, tcp6_echo_server)
  # etc.
```
You can start those from a separate shell with `build/uv_run_tests_a <test> <helper>`.
